### DISCUSSION TITLE
Eager Relations

### DISCUSSION THREAD

**Discussion by robertt** [prisma/prisma#2059] [Q&A] [answered]
## Problem
When loading queries for GraphQL, it's always expected that you have the child data loaded too, so you can query further then the root of whatever you're querying (i.e. you are querying a user, you expect to be able to see their posts etc). The problem is, with prisma you have to specify what relations you want loaded.

## Solution
TypeORM handles this very well with "eager" relations (https://github.com/typeorm/typeorm/blob/master/docs/eager-and-lazy-relations.md). Without thinking too much about it I think an `@eager` "flag" after the `@relation` would be nice.

## Alternatives
You can always just specify every relation when loading it but it's not ideal.

**Comment by schickling** [comment, marked as answer]
Hi @robertwestbury! Thanks a lot for opening this issue/discussion. I unfortunately misread your message and therefore incorrectly turned it into a discussion (which seems like can't be undone at the current point).

I've created a new feature request based on your suggestion: Please feel free to provide more input/feedback over here: https://github.com/prisma/specs/issues/459

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
